bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/internal/framework/processor.aliases

```diff
@@ -33,3 +33,4 @@ s390x                                  # IBM System 390 (64-bit)
 v850e                                   # NEC V850E
 x86-64 amd64 em64t x86_64               # 64 bit x86 architecture
 loongarch64 LoongArch64                 # 64 bit loongarch architecture
+riscv64 				# 64 bit riscv architecture
```

Comment:
FYI. This changes the jgit-based qualifier of the containing bundle and the error is because of this. The version in the MANIFEST.MF should be incremented by +100 on the third segment.

Reply:
Thanks for reviewing this.

hmm, could you help me to identify which MANIFEST.MF file need to be modified? I tried this [one](https://github.com/eclipse-equinox/equinox/pull/669/files#diff-bd841891be612c6b148a8266db6e2c296fe0287f001ad5680f04ec1da50f26cdR6), but get another error. TIA

Reply:
Never mind. I am very appreciate your helping here!

I have changed the bundle version as suggestion above, but I have to modify the [SNAPSHOT version](https://github.com/eclipse-equinox/equinox/pull/669/files#diff-422a3d0e45b0b6a956f405a60efefa68a2f591eb7fd84656d4df1117737eca0aL22) of org.eclipse.osgi to match bundle version also. 

But still failed on my local build:

```
[ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.8:compare-version-with-baselines (compare-attached-artifac
ts-with-release) on project org.eclipse.equinox.core.feature: Only qualifier changed for (org.eclipse.equinox.core.feature.feature.jar/1.15.200.v20240910-2355). Expected to have bigger x.y.z than what is available in baseline (1.15.200.v20240722-2103) -> [Help 1]

```

I notice there is one [forceQualifierUpdate.txt](features/org.eclipse.equinox.core.feature/forceQualifierUpdate.txt) and looking at this can help to fix the issue or not.

Reply:
Sorry, I didn't notice there was a pom.xml.  Most of the projects are pomless now...

The version of this feature needs to be incremented and also the pom.xml of that project:

https://github.com/eclipse-equinox/equinox/pull/669/files#diff-96f61e5538813e844137483f68bb4291675fed70e9eca89215a88495631e2e01

It looks like only the 3rd segment of the version is ever incremented, but I would have expected that adding a fragment would increase the 2nd segment because it's like adding API.  But the guidelines are rather vague on the subject of adding or removing things from a feature:

https://github.com/eclipse-platform/eclipse.platform/blob/master/docs/VersionNumbering.md#Versioning-features

Reply:
Thanks, it seems the issue for version of this feature has been addressed. At least from the current log.

Now I get one not convinced failure like:

```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-packaging-plugin:4.0.8:package-feature (default-package-feature) on project org.eclips
e.equinox.executable: Execution default-package-feature of goal org.eclipse.tycho:tycho-packaging-plugin:4.0.8:package-feature failed: eclips
e-plugin artifact with ID "org.eclipse.equinox.launcher.gtk.linux.riscv64" and version matching "0.0.0" was not found in the target platform
-> [Help 1]
```

I have enabled `org.eclipse.equinox.launcher.gtk.linux.riscv64` bundles on [pom.xml](https://github.com/eclipse-equinox/equinox/pull/669/commits/09f8d33438f022799756cdb0cfdf4a34c57528bb#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8) but It is not triggered by the `mvn clean verify  -DskipTests=true` command. Instead, it is can be built with `./build.sh` manually to generate equinox.launcher.gtk.linux.riscv64 binaries.

Reply:
I think the aggregator's target platform configuration is involved here:

![image](https://github.com/user-attachments/assets/5b950599-2990-43e2-9fff-7e08e00ef772)

I would expect a tuple needs to be added there.  But how to coordinate that with the changes for this PR is a bit of a mystery to me.

@HannesWell  @laeubi @akurtakov 

Guys, any idea how this need to be coordinated?

Reply:
Th environments should actually only matter when packing a product, I would assume that we first need a binary build before one can actual add it to the feature.

Reply:
> I would expect a tuple needs to be added there. But how to coordinate that with the changes for this PR is a bit of a mystery to me.
> 
In https://github.com/eclipse-equinox/equinox/pull/559/files#diff-ed29dcec78f53ccece5e89f61195150d2356f80b793d3368dce3bc4247d0167a we added the new environment temporarily to the `org.eclipse.equinox.executable.feature/pom.xml` until the launcher plus the SWT binaries were in the build. As soon as that is completed the environment can be added globally as you wrote and the temporary addition can be removed again.
We can/should do it this way again to break the chicken-egg-problem and have clean builds all the time.